runtime/conditions/setter.go

```diff
@@ -223,3 +227,15 @@ func hasSameState(i, j *metav1.Condition) bool {
 		i.Reason == j.Reason &&
 		i.Message == j.Message
 }
+
+const maxMessageLength = 32768
+
+// trimConditionMessage trims the condition message to a maximum length of 32K characters.
+func trimConditionMessage(msg string) string {
```

Comment:
Would be good to have some test coverage for this, as the reslicing can later on become a potential victim to out-of-bound errors.

To not have to test the full `maxMessageLength`, you could change it to `trimConditionMessage(msg string, max int)` and do `condition.Message = trimConditionMessage(condition.Message, maxMessageLength)` instead.